pkg/controllers/vdb/onlineupgrade_reconciler.go

```diff
@@ -1016,6 +1111,19 @@ func (r *OnlineUpgradeReconciler) deleteSandboxConfigMap(ctx context.Context) (c
 	return ctrl.Result{}, nil
 }
 
+func (r *OnlineUpgradeReconciler) waitForConnectionRedirect(ctx context.Context) (ctrl.Result, error) {
+	// Iterate through the subclusters in replica group A. We check if there are
+	// any active connections for each. Once they are all idle we can advance to
+	// the next action in the upgrade.
+	for _, scName := range r.VDB.GetSubclustersForReplicaGroup(vmeta.ReplicaGroupAValue) {
+		res, err := r.Manager.isSubclusterIdle(ctx, r.PFacts[vapi.MainCluster], scName)
```

Comment:
Where does the get_drainning_status API come into the picture? Should we use that API to check if any connections are paused in replica A?

Reply:
we could? but it's an idempotent operation, doesn't matter if we just run it again.

Reply:
I just want to understand why we need vcluster get_drainning_status API. In the future, if we use client proxy, do we still need to poll for connection redirect?